scenario_player/node_support.py

```diff
@@ -346,11 +348,26 @@ def api_address(self):
         if not self._api_address:
             self._api_address = self._options.get("api-address")
             if self._api_address is None:
-                # Find a random free port
                 sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-                sock.bind(("127.0.0.1", 0))
-                self._api_address = f"127.0.0.1:{sock.getsockname()[1]}"
-                sock.close()
+                with closing(sock):
+                    # Force the port into TIME_WAIT mode, ensuring that it will not
+                    # be considered 'free' by the OS for the next 60 seconds. This
+                    # does however require that the process using the port sets
+                    # SO_REUSEADDR on it's sockets. Most 'server' applications do.
+                    sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+                    sock.bind(("127.0.0.1", 0))
+
+                    sock_addr = sock.getsockname()
+                    port = sock_addr[1]
+
+                    # Connect to the socket to force it into TIME_WAIT state (see
+                    # above)
+                    sock.listen(1)
+                    sock2 = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
why not use the same function available in the raiden repo? this approach was **not** reliable on CI, which was a cause of flakiness, I'd assume that it will be flaky in the scenario servers too

Reply:
This is only for the raiddit branch.